src/client/opamClient.ml

```diff
@@ -2301,21 +2301,35 @@ let install_t t ?ask ?(ignore_conflicts=false) ?(depext_only=false)
         (OpamSolution.install_depexts ~force_depext:true ~confirm:false t
            (OpamSolver.all_packages solution)), None
       else
-      let add_roots =
-        OpamStd.Option.map (function
-            | true ->
-              OpamPackage.Name.Set.union requested @@
-              OpamPackage.Name.Set.of_list (List.rev_map fst deps_atoms)
-            | false -> OpamPackage.Name.Set.empty)
-          add_to_roots
-      in
-      let t, res =
-        OpamSolution.apply ?ask t
-          ~requested:packages
-          ~print_requested:(print_requested requested formula)
-          ?add_roots ~skip
-          ~download_only ~assume_built solution in
-      t, Some (Success res)
+        let add_roots =
+          match add_to_roots, deps_only with
+          | (Some true | None), true ->
+            let pkgs_solution = OpamSolver.all_packages solution in
+            let requested_deps =
+              OpamPackage.Set.fold (fun nv acc ->
+                  OpamFormula.ors [
+                    OpamPackageVar.all_depends t (OpamSwitchState.opam t nv)
+                      ~depopts:false ~build:true ~post:false;
+                    acc
+                  ])
```

Comment:
No need to call `ors` with only 2 parameters as it will always reduce to `Or (x.(1), x.(2))`
```suggestion
                  OpamFormula.Or (
                    OpamPackageVar.all_depends t (OpamSwitchState.opam t nv)
                      ~depopts:false ~build:true ~post:false,
                    acc))
```

Reply:
true